hazelcast/src/main/java/com/hazelcast/map/impl/querycache/NodeQueryCacheContext.java

```diff
@@ -168,12 +170,22 @@ public Object toObject(Object obj) {
         return mapServiceContext.toObject(obj);
     }
 
-    private String registerLocalIMapListener(String name) {
+    @Override
+    public ContextMutexFactory getLifecycleMutexFactory() {
+        return lifecycleMutexFactory;
+    }
+
+    private String registerLocalIMapListener(final String name) {
         return mapServiceContext.addLocalListenerAdapter(new ListenerAdapter<IMapEvent>() {
             @Override
             public void onEvent(IMapEvent event) {
                 // NOP
             }
+
+            @Override
+            public String toString() {
+                return "Local IMap listener for '" + name + "' map";
```

Comment:
Would change this to `"Local IMap listener for map '" + name + "'"` (sounds a bit smoother and user defined map names could be very long)

Reply:
fixed